### DISCUSSION TITLE
load csv data with double values into a decimal column

### DISCUSSION THREAD

**Discussion by peeyee** [ClickHouse/ClickHouse#50172] [Q&A] [answered]
Here, I load csv data into clickhouse, and got a double value, like 3.1415926。 when I insert it into a decimal(12,4) column, the default behavior is truncate: drop the rest number , and put 3.1415 into the column。
However，I wish the value to be 3.1416(like use round), how could I do this? 
many thanks!

**Comment by DanRoscigno** [comment, marked as answer]
Try this:
```sql
CREATE TABLE load_csv
(
    `name` String,
    `num` Decimal(12, 4)
)
ENGINE = MergeTree
ORDER BY name;
```

CSV File `data.csv`
```
test, 3.1415926
```

```bash
cat data.csv | clickhouse local --table='incoming_csv_data' --input-format='CSVWithNames' --query="select name, round(num, 4) from incoming_csv_data" | clickhouse client --ask-password --query='INSERT INTO load_csv FORMAT TSV'
```

Notes: There may be an easier way, this is what I am used to:
- cat the file
- preprocess with clickhouse local
- pipe to clickhouse client

The `FORMAT TSV` at the end might be confusing, that is there because the default output format for local is TSV
